components/usage/pkg/stripe/stripe.go

```diff
@@ -122,6 +130,38 @@ func (c *Client) updateUsageForCustomer(ctx context.Context, customer *stripe.Cu
 	}, nil
 }
 
+// GetUpcomingInvoice fetches the upcoming invoice for the given team
+func (c *Client) GetUpcomingInvoice(ctx context.Context, teamId string) (*StripeInvoice, error) {
+	query := fmt.Sprintf("metadata['teamId']:'%s'", teamId)
+	searchParams := &stripe.CustomerSearchParams{
+		SearchParams: stripe.SearchParams{
+			Query:   query,
+			Expand:  []*string{stripe.String("data.subscriptions")},
+			Context: ctx,
+		},
+	}
+	iter := c.sc.Customers.Search(searchParams)
+	if !iter.Next() {
```

Comment:
First check for errors:
```suggestion
	if iter.Error() != nil {
```